### PR TITLE
[CoreAudio] Update bindings up to Xcode 27.0 Beta 3

### PR DESCRIPTION
**Pull request by dalexsoto**
## Summary

- Bind `AudioServerPlugInRegisterMediaDeviceExtension` for Mac Catalyst 27.0 as `AudioToolbox.AudioServerPlugIn.RegisterMediaDeviceExtension`.
- Marshal the nullable interruption handler with the established `BlockLiteral` callback pattern.
- Document the opaque native plug-in lifetime and remote-device restrictions.
- Add a Mac Catalyst runtime test covering non-null and null interruption handlers, and remove the resolved CoreAudio xtro todo.

The binding is intentionally limited to Mac Catalyst because this is the only platform reported by the generated CoreAudio binding inventory and corresponding xtro todo.

## Validation

- `make world`
- Clean xtro generation/classification: sanity passed
- Clean cecil test suite: passed
- Mac Catalyst introspection:
  - P/Invoke fixture: 8/8 passed
  - Signature fixture: 3/3 passed
- Updated `AudioServerPlugInTest`: builds successfully and is correctly ignored on the local macOS 26.5 host; macOS 27 CI executes both native-call paths
- tvOS introspection: 43/43 passed
- macOS introspection: 32/32 passed, with 2 expected host-version ignores
- iOS focused introspection:
  - P/Invoke fixture: 8/8 passed
  - Signature fixture: 3/3 passed

The app-size fixture discovered all 16 cases, which were intentionally skipped by the existing Xcode-beta guard. Broader monotouch runs only reported unrelated AVAudioSourceNode simulator callback-count failures and transient network timeouts. The full iOS introspection run was limited by the Xcode 27 simulator scene watchdog, and the full Mac Catalyst constructor run hit the known unsigned-app TCC abort in `CLLocationButton`; the binding-relevant fixtures above passed independently.

### CODE REVIEW

File: tests/monotouch-test/AudioToolbox/AudioServerPlugInTest.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class AudioServerPlugInTest {
+
+		[Test]
+		public void RegisterMediaDeviceExtensionSignature ()
+		{
+			Func<IntPtr, Action?, int> method = AudioServerPlugIn.RegisterMediaDeviceExtension;
+
+			Assert.That (method, Is.Not.Null);
```
Comment (rolfbjarne) [inline tests/monotouch-test/AudioToolbox/AudioServerPlugInTest.cs]: That's a rather weak test... it just asserts that the method exists.

Would it work to call the method?

```cs
AudioServerPlugIn.RegisterMediaDeviceExtension (IntPtr.Zero, () => {});
AudioServerPlugIn.RegisterMediaDeviceExtension (IntPtr.Zero, null);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tests/monotouch-test/AudioToolbox/AudioServerPlugInTest.cs (hunk 1 above)
Comment (dalexsoto) [reply tests/monotouch-test/AudioToolbox/AudioServerPlugInTest.cs]: Fixed in 078b412455. The test now calls `RegisterMediaDeviceExtension` with both a non-null callback and a null callback. It is gated with `TestRuntime.AssertSystemVersion (ApplePlatform.MacCatalyst, 27, 0)` because the symbol is not exported by the macOS 26.5 host CoreAudio framework; the focused test builds and reports the expected ignore locally, while the macOS 27 CI host will execute both native-call paths.

**Review by rolfbjarne — APPROVED** [review]
